tests/testsuite/build_script.rs

```diff
@@ -5828,6 +5828,25 @@ fn links_overrides_with_target_applies_to_host() {
         .with_status(0)
         .run();
 
+    // The name is different on windows because ".dll" is not automatically
+    // appenended like the extension on other OSes is, despite being needed.
+
+    // The path is different on windows because we need the "name.dll.lib" file
+    // to link to it, and as per #3934 that only lives in `target/{profile}/deps`
+    // and isn't copied to `target/{profile}` like the .dll is. #3951 removed the
+    // hash mentioned in #3934 but did not change the directory of the .lib file.
+
+    // Meanwhile on other operating systems everything we need to link (which is
+    // just the linux .so/the macos .dylib) is copied to `target/{profile}` from
+    //  `deps` and as the blessed output artifact that seems to be the appropriate
+    // version to use.
+    let mylib_name = if cfg!(windows) { "mylib.dll" } else { "mylib" };
```

Comment:
Instead of doing this, perhaps we can go an easier route like other tests: Asserting rustflags for rustc invocation?

https://github.com/rust-lang/cargo/blob/6d685637f1853b705053500ff4abacdc51df9c91/tests/testsuite/build_script.rs#L1653-L1670

Reply:
This definitely looks better, and is exactly the pre-existing test for link overrides that I failed to find. Will do this.